### PR TITLE
[guiinfo][PVR] Fix PVR.EpgeventProgress GUI int & some GUIInfoManager cleanup

### PR DESCRIPTION
**Pull request by ksooo**
First commit fixes a regression reported here: https://forum.kodi.tv/showthread.php?tid=298462&pid=2708380#pid2708380

The other commits are all `CGUIInfoManager` cleanups.

@Jalle19 mind taking a look?

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -9451,18 +9432,13 @@ bool CGUIInfoManager::GetItemInt(int &value, const CGUIListItem *item, int info)
...
       {
         const CFileItem *pItem = static_cast<const CFileItem*>(item);
-        if (pItem && pItem->HasPVRChannelInfoTag())
-        {
-          CPVREpgInfoTagPtr epgNow(pItem->GetPVRChannelInfoTag()->GetEPGNow());
-          if (epgNow)
-            value = (int) epgNow->ProgressPercentage();
-        }
-        else if (pItem && pItem->HasEPGInfoTag())
+        if (pItem->IsPVRChannel() || pItem->IsEPG())
```
Comment (Jalle19) [inline xbmc/GUIInfoManager.cpp]: There used to be a null check here, did you remove it intentionally?

**Review by Jalle19 — APPROVED** [review]

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/GUIInfoManager.cpp]: Yes, this is by intention. If `item->IsFileItem()` returns `true`, the cast `const CFileItem *pItem = static_cast<const CFileItem*>(item);` must succeed.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (Jalle19) [reply xbmc/GUIInfoManager.cpp]: Cool

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**